### PR TITLE
expose loglevel and dask failure mode on CLI

### LINKED ISSUE #784: Consider changing default log level to be higher and expose on CLI
Currently in the workflows the default loglevel is `DEBUG` which is causing an avalanche of information. We should also expose this as a CLI option also.

### LINKED ISSUE #817: Expose dask failure mode on CLI
Because Dask is failed compute tolerant, it can sometimes be hard to debug the dask code paths as errors are not well captured in `error="skip"` mode. We should expose this as a global CLI variable.

### PR DESCRIPTION
**Pull request by hmacdope**
## Description
Fixes #817 
Fixes #784 

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@kaminow @apayne97 @JenkeScheen @jthorton can I grab a review here? :)

File: asapdiscovery-cli/asapdiscovery/cli/cli_args.py [hunk 1]
```diff
@@ -54,8 +54,17 @@ def dask_type(func):
     )(func)
 
 
+def dask_failure_mode(func):
+    return click.option(
+        "--dask-failure-mode",
+        type=click.Choice(DaskFailureMode.get_values(), case_sensitive=False),
+        default=DaskFailureMode.SKIP,
+        help="The failure mode for dask. Can be 'raise' or 'skip'.",
```
Comment (jthorton) [inline asapdiscovery-cli/asapdiscovery/cli/cli_args.py:62]: maybe we should add show default to make it clear if the user needs to use the option or not?

File: asapdiscovery-cli/asapdiscovery/cli/cli_args.py [hunk 2]
```diff
@@ -226,6 +225,15 @@ def save_to_cache(func):
     )(func)
 
 
+def loglevel(func):
+    return click.option(
+        "--loglevel",
+        type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+        help="The log level to use.",
+        default="INFO",
```
Comment (jthorton) [inline asapdiscovery-cli/asapdiscovery/cli/cli_args.py:233]: same here add `show_default=True`

**Review by jthorton — APPROVED** [review]
Great addition, very minor suggestions feel free to merge once changed.

**Pull request merged**